lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1005,7 +1005,7 @@ bool TypeChecker::typeCheckCatchPattern(CatchStmt *S, DeclContext *DC) {
 
 static bool isDiscardableType(Type type) {
   return (type->is<ErrorType>() ||
-          type->isNever() ||
+          type->isUninhabited() ||
```

Comment:
We should except in specific situations (e.g. when optional chaining a call to a void-returning method)